packages/ai/src/generate-image/generate-image.ts

```diff
@@ -192,10 +187,23 @@ Only applicable for HTTP-based providers.
       for (const [providerName, metadata] of Object.entries<{
         images: unknown;
       }>(result.providerMetadata)) {
-        providerMetadata[providerName] ??= { images: [] };
-        providerMetadata[providerName].images.push(
-          ...result.providerMetadata[providerName].images,
-        );
+        if (providerName === 'gateway') {
+          providerMetadata[providerName] = {
+            ...metadata,
+          } as ImageModelV2ProviderMetadata[string];
+          const imagesValue = (
+            providerMetadata[providerName] as { images?: unknown }
+          ).images;
+          if (Array.isArray(imagesValue) && imagesValue.length === 0) {
+            delete (providerMetadata[providerName] as { images?: unknown })
+              .images;
+          }
+        } else {
+          providerMetadata[providerName] ??= { images: [] };
+          providerMetadata[providerName].images.push(
+            ...result.providerMetadata[providerName].images,
+          );
+        }
```

Comment:
Sorry I've been starring at this code and I don't get it yet.

I understand that gateway returns request-specific metadata. But a request can yield more than one images, including with gateway, correct? And each of these images might yield their own meta data?

What confuses me is that for gateway images, the code still iterates through all images, and gets metadata for each image, but then ends up merging the metadata from all images into the single `providerMetadata['gateway']` object?